### PR TITLE
ci: Add PagerDuty integration to XTS suite and promote build candidate

### LINKED ISSUE #16730: CITR Phase 1: Set up PagerDuty Rotations
- Enable pager duty alerts for CITR
- Set up rotation schedule for daytime hours

### PR DESCRIPTION
**Pull request by rbarkerSL**
**Description**:

- Adds PagerDuty integration to XTS and Promote Build Candidate workflows
- Resolves outstanding CITR phase 1 action

**Related Issue(s)**:
Fixes #16730

### CODE REVIEW

File: .github/workflows/zxcron-extended-test-suite.yaml [hunk 1]
```diff
@@ -285,6 +285,31 @@ jobs:
...
+      - name: Report failure (PagerDuty)
+        run: |
+          curl --request 'POST' \
+          --url 'https://events.pagerduty.com/v2/enqueue' \
+          --header 'Content-Type: application/json' \
+          --data '{
+            "payload": {
+              "summary": "Hedera Services - eXtended Test Suite Failure",
+              "severity": "error",
+              "source": "GitHub Actions - Extended Test Suite",
```
Comment (andrewb1269hg) [inline .github/workflows/zxcron-extended-test-suite.yaml:297]: Should this be: `eXtended Test Suite` to match line 295 syntax?

File: .github/workflows/zxcron-promote-build-candidate.yaml [hunk 2]
```diff
@@ -230,6 +230,27 @@ jobs:
...
+      - name: Report failure (PagerDuty)
+        run: |
+          curl --request 'POST' \
+          --url 'https://events.pagerduty.com/v2/enqueue' \
+          --header 'Content-Type: application/json' \
+          --data '{
+            "payload": {
+              "summary": "Hedera Services - eXtended Test Suite Failure",
+              "severity": "error",
+              "source": "GitHub Actions - Extended Test Suite",
```
Comment (andrewb1269hg) [inline .github/workflows/zxcron-promote-build-candidate.yaml:242]: Should this be: eXtended Test Suite to match line 240 syntax?

File: .github/workflows/zxcron-extended-test-suite.yaml [hunk 3]
```diff
@@ -285,6 +285,31 @@ jobs:
...
+                "Execute eXtended Test Suite": "${{ needs.extended-test-suite.result }}",
+                "JRS Panel": "${{ needs.abbreviated-panel.result }}",
+                "Hedera Node JRS Panel": "${{ needs.hedera-node-jrs-panel.result }}",
+                "Tag as XTS-Passing": "${{ needs.tag-for-promotion.result }}",
+                "Source Commit": "<${{ github.server_url }}/${{ github.repository }}/commit/${{ needs.fetch-xts-candidate.outputs.xts-tag-commit }}>",
+                "Commit Author": "${{ needs.fetch-xts-candidate.outputs.xts-tag-commit-author }}",
+                "Workflow Run": "<${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}>"
+              }
+            },
+            "routing_key": "${{ secrets.PAGERDUTY_CITR_INTEGRATION_KEY }}",
```
Comment (andrewb1269hg) [inline .github/workflows/zxcron-extended-test-suite.yaml:309]: I don't see this key passed into the workflow. Does it need to be or is it an env token available implicitly to the whole workflow?

File: .github/workflows/zxcron-promote-build-candidate.yaml [hunk 4]
```diff
@@ -230,6 +230,27 @@ jobs:
...
+              "custom_details": {
+                "Fetch Latest Build Candidate": "${{ needs.determine-build-candidate.result }}",
+                "Promote Build Candidate": "${{ needs.promote-build-candidate.result }}",
+                "Source Commit": "<${{ github.server_url }}/${{ github.repository }}/commit/${{ github.sha }}>",
+                "Workflow Run": "<${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}>"
+              }
+            },
+            "routing_key": "${{ secrets.PAGERDUTY_CITR_INTEGRATION_KEY }}",
```
Comment (andrewb1269hg) [inline .github/workflows/zxcron-promote-build-candidate.yaml:250]: Same as above - I don't see this key passed into the workflow. Does it need to be or is it an env token available implicitly to the whole workflow?

**Review by andrewb1269hg — CHANGES_REQUESTED** [review]

File: .github/workflows/zxcron-extended-test-suite.yaml (hunk 3 above)
Comment (rbarkerSL) [reply .github/workflows/zxcron-extended-test-suite.yaml:309]: It's a workflow dispatch trigger and a crib trigger. These have access to the repository secrets implicitly

File: .github/workflows/zxcron-promote-build-candidate.yaml (hunk 4 above)
Comment (rbarkerSL) [reply .github/workflows/zxcron-promote-build-candidate.yaml:250]: It's a workflow dispatch trigger and a crib trigger. These have access to the repository secrets implicitly

File: .github/workflows/zxcron-extended-test-suite.yaml (hunk 1 above)
Comment (rbarkerSL) [inline .github/workflows/zxcron-extended-test-suite.yaml:297]:
```suggestion
              "source": "GitHub Actions - eXtended Test Suite",
```

File: .github/workflows/zxcron-promote-build-candidate.yaml (hunk 2 above)
Comment (rbarkerSL) [inline .github/workflows/zxcron-promote-build-candidate.yaml:242]:
```suggestion
              "source": "GitHub Actions - eXtended Test Suite",
```

**Review by andrewb1269hg — APPROVED** [review]
Looks great!

**Pull request merged**